rasa/core/processor.py

```diff
@@ -324,22 +322,53 @@ def _has_message_after_reminder(
             or not self._is_reminder_still_valid(tracker, reminder_event)
         ):
             logger.debug(
-                f"Canceled reminder because it is outdated. "
-                f"(event: {reminder_event.action_name} id: {reminder_event.name})"
+                f"Canceled reminder because it is outdated. " f"({str(reminder_event)})"
             )
         else:
-            # necessary for proper featurization, otherwise the previous
-            # unrelated message would influence featurization
-            tracker.update(UserUttered.empty())
-            action = self._get_action(reminder_event.action_name)
-            should_continue = await self._run_action(
-                action, tracker, output_channel, nlg
+            intent = reminder_event.intent
+            entities = reminder_event.entities or {}
+            await self.trigger_external_user_uttered(
+                intent, entities, tracker, output_channel
             )
-            if should_continue:
-                user_msg = UserMessage(None, output_channel, sender_id)
-                await self._predict_and_execute_next_action(user_msg, tracker)
-            # save tracker state to continue conversation from this state
-            self._save_tracker(tracker)
+
+    async def trigger_external_user_uttered(
+        self,
+        intent_name: Text,
+        entities: Optional[Union[List[Dict[Text, Any]], Dict[Text, Text]]],
+        tracker: DialogueStateTracker,
+        output_channel: OutputChannel,
+    ) -> None:
+        """Triggers an external message.
+
+        Triggers an external message (like a user message, but invisible;
+        used, e.g., by a reminder or the trigger_intent endpoint).
+
+        Args:
+            intent_name: Name of the intent to be triggered.
+            entities: List of entities to be passed on.
```

Comment:
since it also can be a `dict`:
```suggestion
            entities: Entities to be passed on.
```